agent/core/src/main/java/org/jolokia/discovery/MulticastUtil.java

```diff
@@ -17,22 +18,40 @@
  */
 public final class MulticastUtil {
 
-    // IPv4 Address for Jolokia's Multicast group
-    public static final String JOLOKIA_MULTICAST_GROUP = "239.192.48.84";
+    // Default IPv4 Address for Jolokia's Multicast group
+    public static final String DEFAULT_JOLOKIA_MULTICAST_GROUP = "239.192.48.84";
 
-    // Multicast port where to listen for queries
-    public static final int JOLOKIA_MULTICAST_PORT = 24884;
+    // Default multicast port where to listen for queries
+    public static final int DEFAULT_JOLOKIA_MULTICAST_PORT = 24884;
 
     // Utility class
     private MulticastUtil() {
     }
 
+    private static <T> T getOrDefault(ConfigKey configKey, String sysEnvKey, T defaultValue) {
+        String property = System.getProperty("jolokia." + configKey.getKeyValue());
```

Comment:
This lookup is already performed upwards in the stack and shouldn't happen here.